### PR TITLE
[receiver/hostmetrics] Add filesystem utilization

### PR DESCRIPTION
**Pull request by brushknight**
**Description:** <Describe what has changed.>
- This PR adds `system.filesystem.utilization` to `hostmetricsreceiver`

**Link to tracking Issue:**
- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6221

**Testing:** 
- Current tests were updated
- Did a sanity check with Ubuntu VM

**Documentation:** 
- Added to scrapper docs.

### CODE REVIEW

File: receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/metadata.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ attributes:
 
   state:
     description: Breakdown of filesystem usage by type.
-    enum: [free, reserved, used]
+    enum: [free, reserved, used, utilized]
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/metadata.yaml:15]: I don't think `utilized` state is needed. `state` attribute looks redundant for the utilization metric, it always has the same value

Comment (brushknight) [reply receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/metadata.yaml:15]: What should be used then? Could it be just an empty string?

Comment (brushknight) [reply receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/metadata.yaml]: done

File: receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/metadata.yaml [hunk 2]
```diff
@@ -37,3 +37,11 @@ metrics:
       aggregation: cumulative
       monotonic: false
     attributes: [device, mode, mountpoint, type, state]
+
+  system.filesystem.utilization:
+    enabled: false
+    description: Percentage of filesystem bytes used.
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/metadata.yaml:43]: According to 
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/README.md#instrument-naming 

> utilization - an instrument that measures the fraction of usage out of its limit should be called entity.utilization. For example, system.memory.utilization for the fraction of memory in use. Utilization values are in the range [0, 1].

So this metrics hast to be fraction of 1 and cannot be percentage unless you want to propose a change to the spec

Comment (brushknight) [reply receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/metadata.yaml:43]: Good point, it is actually 0 -> 1 float value. I will update description 👍

Comment (brushknight) [reply receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/metadata.yaml]: Changed to `Fraction of filesystem bytes used.`

File: receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper_others.go [hunk 3]
```diff
@@ -28,9 +28,19 @@ const fileSystemStatesLen = 2
...
+			deviceUsage.partition.Mountpoint, deviceUsage.partition.Fstype,
+			metadata.AttributeState.Used)
 		s.mb.RecordSystemFilesystemUsageDataPoint(
-			now, int64(deviceUsage.usage.Free), deviceUsage.partition.Device, getMountMode(deviceUsage.partition.Opts), deviceUsage.partition.Mountpoint, deviceUsage.partition.Fstype, metadata.AttributeState.Free)
+			now, int64(deviceUsage.usage.Free),
+			deviceUsage.partition.Device, getMountMode(deviceUsage.partition.Opts),
+			deviceUsage.partition.Mountpoint, deviceUsage.partition.Fstype,
+			metadata.AttributeState.Free)
+		s.mb.RecordSystemFilesystemUtilizationDataPoint(
+			now, deviceUsage.usage.UsedPercent,
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper_others.go:41]: I think `usage.UsedPercent` is still reported in percents

Comment (brushknight) [reply receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper_others.go:41]: You are right, I will fix it. Thank you

File: CHANGELOG.md [hunk 4]
```diff
@@ -22,6 +22,7 @@
 
 ### 💡 Enhancements 💡
 
+- `hostreceiver/filesystemscraper`: Add filesystem utilization (#8027)
```
Comment (dmitryax) [inline CHANGELOG.md:25]: Please move this line to Unreleased

**Review by dmitryax — COMMENTED** [review]
One nit, otherwise LGTM

**Comment by brushknight** [comment]
Change log conflicts appear every few hours, will resolve it right before merge

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
Sorry for the delay. LGTM

**Pull request merged**